### DISCUSSION TITLE
Open in new tab

### DISCUSSION THREAD

**Discussion by jwillmer** [mkdocs/mkdocs#2704] [Ideas]
Is it possible to open a page in a new tab? I wrote my use case to the [nav-plugin repository](https://github.com/oprypin/mkdocs-literate-nav/issues/7) but got redirected to this discussion thread.

I basically would like to define which pages should open in a new tab like so:

`.pages`
```
nav:
  - Page 1: page1.md#open-in-new-tab
  - Page 2: page2.md
```

I saw there are possibilities to open links in a document in a new tab with the help of markdown. Maybe the same feature could be made available in the `.pages` file?

**Comment by oprypin** [comment]
This is a slightly hacky way because it rewrites the final HTML, but it works:

```diff
diff --git a/hooks.py b/hooks.py
new file mode 100644
index 000000000..2ebe3f88d
--- /dev/null
+++ b/hooks.py
@@ -0,0 +1,8 @@
+import re
+
+def on_post_page(output, **kwargs):
+    return re.sub(
+        r'href="([^"]+)#open-in-new-tab"',
+        r'href="\1" target="_blank"',
+        output
+    )
diff --git a/mkdocs.yml b/mkdocs.yml
index bfaeb5588..b33073cf1 100644
--- a/mkdocs.yml
+++ b/mkdocs.yml
@@ -38,0 +39 @@ nav:
+    - Test: https://example.org/#open-in-new-tab
@@ -61,0 +63,3 @@ plugins:
+    - mkdocs-simple-hooks:
+        hooks:
+          on_post_page: 'hooks:on_post_page'
```

**Reply by jwillmer** [reply]
Took me a moment to realize that you suggested to use an [existing plugin](https://github.com/aklajnert/mkdocs-simple-hooks). But it works perfectly :)

**Comment by squidfunk** [comment]
Just for the sake of completeness – you can use the [Attribute Lists](https://python-markdown.github.io/extensions/attr_list/) extension to achieve the same:

``` markdown
[Title](url){ target="_blank" }
```

**Reply by jwillmer** [reply]
I don't think this is working on `.pages`.

**Reply by squidfunk** [reply]
Ah sorry, you're right. Nevermind then. However, maybe it's useful for others that search for "open in new tab". This is at least the canonical way to do it from within Markdown 😊
